File: backend/app/utils/bet_calculator.py
def calculate_payout(bet_amount: float, odds: float) -> float:
    """Calculate total payout (stake + profit) for American odds."""
    if odds > 0:
        # Positive odds: profit = stake * (odds / 100)
        profit = bet_amount * (odds / 100)
    else:
        # Negative odds: profit = stake * (100 / abs(odds))
        profit = bet_amount * (100 / abs(odds))
    
    return bet_amount + profit

File: frontend/src/utils/betCalculator.js
export function calculatePayout(betAmount, odds) {
    const amount = parseFloat(betAmount)
    if (isNaN(amount) || amount <= 0) return 0

    let profit
    if (odds > 0) {
        profit = amount * (odds / 100)
    } else {
        profit = amount * (100 / Math.abs(odds))
    }

    return parseFloat((amount + profit).toFixed(2))
}